dask_sql/physical/rex/core/call.py

```diff
@@ -682,24 +682,67 @@ def __init__(self):
         super().__init__(self.timestampadd)
 
     def timestampadd(self, unit, interval, df: SeriesOrScalar):
-        df = convert_to_datetime(df)
-
-        if unit in {"DAY", "SQL_TSI_DAY"}:
-            return df + np.timedelta64(interval, "D")
-        elif unit in {"HOUR", "SQL_TSI_HOUR"}:
-            return df + np.timedelta64(interval, "h")
-        elif unit == "MICROSECOND":
-            return df + np.timedelta64(interval, "us")
-        elif unit == "MILLISECOND":
-            return df + np.timedelta64(interval, "ms")
-        elif unit in {"MINUTE", "SQL_TSI_MINUTE"}:
-            return df + np.timedelta64(interval, "m")
-        elif unit in {"SECOND", "SQL_TSI_SECOND"}:
-            return df + np.timedelta64(interval, "s")
-        elif unit in {"WEEK", "SQL_TSI_WEEK"}:
-            return df + np.timedelta64(interval * 7, "W")
+        unit = unit.upper()
+        interval = int(interval)
+        if interval < 0:
+            raise RuntimeError(f"Negative time interval {interval} is not supported.")
+        df = df.astype("datetime64[ns]")
+
+        if unit in {"YEAR", "YEARS"}:
+            result = []
+            for date in df:
+                year = date.year + interval
+                # Check leap day
+                if year % 4 != 0 and date.month == 2 and date.day == 29:
+                    result.append(date.replace(year=year, month=3, day=1))
+                else:
+                    result.append(date.replace(year=year))
+            return pd.Series(result)
```

Comment:
Is there an alternative approach that can achieve this without iterating over each row individually and specifically return a pandas series?